packages/fastify-api-reference/vite.config.ts

```diff
@@ -2,7 +2,7 @@ import { viteStaticCopy } from 'vite-plugin-static-copy'
 import { defineConfig } from 'vitest/config'
 
 import pkg from './package.json'
-import { nodeExternals } from './vite-plugins'
+import { nodeExternals } from './vite-plugins/nodeExternals.js'
```

Comment:
Regarding the `.js` extensions, you can set `"allowImportingTsExtensions": true` in your `tsconfig.node.json` to allow using explicit `.ts` extensions in the code. You might prefer that, based on what you wrote in the PR description :)

Reply:
I believe allowImportingTsExtensions requires moduleResolution=bundler and I think this PR changes the resolution to NodeNext. However, in TS 5.7 `rewriteRelativeImportExtensions` was introduced for this reason (inability to use .ts extensions for code that actually produces JS output). Assuming it might be out of scope to bump the TS version in this change, though

Reply:
Ah, that's good to know!

Reply:
@chrisbenincasa we can definitely put up another PR to bump the TS version!

Reply:
@Greenheart something else must be at work there - it doesn't seem to be invalid, but vscode warns against this explicitly: 

![image](https://github.com/user-attachments/assets/ca45dd25-b22c-418e-a20d-d437e630705f)

but interestingly the docs only mention the emit parts: https://www.typescriptlang.org/tsconfig/#allowImportingTsExtensions

So perhaps it won't cause outright failure.

Reply:
Yeah, it’s all not great. I think I’ll just see if I can get used to it.

Reply:
there's gotta be another way! Can't we just do it in the build, so it builds out a NodeNext version?